`nextflow module run` prints `[SUCCESS]` even when execution fails with exit code 1

## Bug Report

### Description

When running `nextflow module run` and a process fails (non-zero exit status), the summary line still prints `[SUCCESS]` before the process exits with code 1. This is confusing and contradicts the `[ERROR]` and `[FAILED]` messages printed above it.

### Example

```bash
nextflow module run nf-core/salmon/index \
    --genome_fasta genome.fa \
    --transcript_fasta transcriptome.fa \
    --outdir results_salmon_index \
    -ansi-log false
```

### Output

```
[PIPELINE] main.nf | profile=standard
[WORKDIR] /Users/pditommaso/Projects/nextflow/test-modules/work
[PROCESS 0d/40e93f] SALMON_INDEX (transcriptome.fa)
[ERROR] ERROR ~ Error executing process > 'SALMON_INDEX (transcriptome.fa)'

Caused by:
  Process `SALMON_INDEX (transcriptome.fa)` terminated with an error exit status (1)

...

[ERROR] SALMON_INDEX (transcriptome.fa)
exit: 1
...
[ERROR] ERROR ~ Cannot cast object 'DataflowBroadcast around DataflowStream[?]' ...

[FAILED] completed=1 failed=1 cached=0
```

The exit code is 1, but another run that also failed showed:

```
[SUCCESS] completed=0 failed=0 cached=0
```

This happens when the failure occurs before task execution (e.g. missing required parameters) or when the DataflowBroadcast error aborts the session after the task runs. In both cases the summary says `[SUCCESS]` despite the command exiting with code 1.

### Expected behavior

The summary line should reflect the actual outcome:
- `[SUCCESS]` only when all tasks complete without errors and the runner exits with code 0
- `[FAILED]` or `[ERROR]` when any task fails or the runner encounters an error

### Environment

- Nextflow version: 26.03.0-edge
- Java version: OpenJDK 21+35
- OS: macOS 26.3.1

--- a/modules/nextflow/src/main/groovy/nextflow/trace/AgentLogObserver.groovy
+++ b/modules/nextflow/src/main/groovy/nextflow/trace/AgentLogObserver.groovy
@@ -220,7 +220,8 @@ class AgentLogObserver implements TraceObserverV2, LogObserver {
         def cached = stats?.cachedCount ?: 0
         def completed = succeeded + failed
 
-        def status = failed > 0 ? 'FAILED' : 'SUCCESS'
+        def hasSessionError = session && !session.isSuccess()
+        def status = (failed > 0 || hasSessionError) ? 'FAILED' : 'SUCCESS'
         printLine("\n[${status}] completed=${completed} failed=${failed} cached=${cached}")
     }
 

--- a/modules/nextflow/src/test/groovy/nextflow/trace/AgentLogObserverTest.groovy
+++ b/modules/nextflow/src/test/groovy/nextflow/trace/AgentLogObserverTest.groovy
@@ -103,8 +103,12 @@ class AgentLogObserverTest extends Specification {
         def statsObserver = Stub(WorkflowStatsObserver) {
             getStats() >> stats
         }
+        def session = Stub(Session) {
+            isSuccess() >> true
+        }
         def observer = new TestAgentLogObserver(output)
         observer.setStatsObserver(statsObserver)
+        observer.onFlowCreate(session)
 
         when:
         observer.onFlowComplete()
@@ -123,14 +127,42 @@ class AgentLogObserverTest extends Specification {
         def statsObserver = Stub(WorkflowStatsObserver) {
             getStats() >> stats
         }
+        def session = Stub(Session) {
+            isSuccess() >> false
+        }
         def observer = new TestAgentLogObserver(output)
         observer.setStatsObserver(statsObserver)
+        observer.onFlowCreate(session)
 
         when:
         observer.onFlowComplete()
 
         then:
         output[0] == '\n[FAILED] completed=7 failed=2 cached=0'
+    }
+
+    def 'should output failed summary when session has error but no task failures'() {
+        given:
+        def output = []
+        def stats = new WorkflowStats()
+        stats.succeededCount = 0
+        stats.failedCount = 0
+        stats.cachedCount = 0
+        def statsObserver = Stub(WorkflowStatsObserver) {
+            getStats() >> stats
+        }
+        def session = Stub(Session) {
+            isSuccess() >> false
+        }
+        def observer = new TestAgentLogObserver(output)
+        observer.setStatsObserver(statsObserver)
+        observer.onFlowCreate(session)
+
+        when:
+        observer.onFlowComplete()
+
+        then:
+        output[0] == '\n[FAILED] completed=0 failed=0 cached=0'
     }
 
     def 'should handle task complete for failed task'() {

